src/session_impl.cpp

```diff
@@ -1887,6 +1864,50 @@ namespace {
 		}
 	}
 
+	void session_impl::remove_listen_sockets(std::vector<listen_endpoint_t>& eps)
+	{
+		m_listen_sockets.remove_if([&eps, this](listen_socket_t const& sock)
```

Comment:
this will only consolidate the elements not removed in the beginning of the container, and return an iterator to the first element to erase. you still need to call ``m_listen_sockets.erase(iter, m_listen_sockets.end())``, or am I missing something subtle?